src/Time/Timestamp.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE Rank2Types                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+
+module Time.Timestamp
+       ( Timestamp (..)
+       , timeDiff
+       , timeAdd
+       , timeMul
+       , timeDiv
+       ) where
+
+import GHC.Real (denominator, numerator, (%))
+
+import Time.Rational (KnownRat, RatioNat)
+import Time.Units (Time (..))
+
+-- | Similar to 'Time' but has no units and can be negative.
+newtype Timestamp = Timestamp Rational
+    deriving (Show, Read, Num, Eq, Ord, Enum, Fractional, Real, RealFrac)
+
+
+-- | Returns the result of comparison of two 'Timestamp's and
+-- the 'Time' of that difference of given time unit.
+timeDiff :: KnownRat unit => Timestamp -> Timestamp -> (Ordering, Time unit)
+timeDiff (Timestamp a) (Timestamp b) =
+    let order         = compare a b
+        dif           = abs (a - b)
+        t :: RatioNat = fromIntegral (numerator dif) % fromIntegral (denominator dif)
```

Comment:
`dif` is already normalized because `(-)` operator for `Rational` performs normalization. So you can use just `(:%)` here.